spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/sql/SparkExecutionPlanProcessor.scala

```diff
@@ -17,25 +17,44 @@
 
 package com.hortonworks.spark.atlas.sql
 
+import java.util.concurrent.TimeUnit
+
 import com.hortonworks.spark.atlas.sql.CommandsHarvester.WriteToDataSourceV2Harvester
 import com.hortonworks.spark.atlas.sql.SparkExecutionPlanProcessor.SinkDataSourceWriter
-
 import org.apache.spark.sql.execution._
 import org.apache.spark.sql.execution.command._
 import org.apache.spark.sql.execution.datasources.{InsertIntoHadoopFsRelationCommand, SaveIntoDataSourceCommand}
 import org.apache.spark.sql.execution.datasources.v2.WriteToDataSourceV2Exec
 import org.apache.spark.sql.execution.streaming.sources.MicroBatchWriter
 import org.apache.spark.sql.hive.execution._
 import org.apache.spark.sql.sources.v2.writer.{DataWriterFactory, WriterCommitMessage}
-import com.hortonworks.spark.atlas.{AbstractEventProcessor, AtlasClient, AtlasClientConf}
+import com.hortonworks.spark.atlas.{AbstractEventProcessor, AtlasClient, AtlasClientConf, AtlasUtils}
 import com.hortonworks.spark.atlas.utils.Logging
 import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.streaming.StreamExecution
 import org.apache.spark.sql.sources.v2.writer.streaming.StreamWriter
 import org.apache.spark.sql.streaming.SinkProgress
+import org.apache.spark.sql.streaming.StreamingQueryListener.QueryProgressEvent
 
 
-case class QueryDetail(qe: QueryExecution, executionId: Long,
-  executionTime: Long, query: Option[String] = None, sink: Option[SinkProgress] = None)
+case class QueryDetail(
+    qe: QueryExecution,
+    executionId: Long,
+    durationMs: Long,
+    query: Option[String] = None,
+    sink: Option[SinkProgress] = None)
+
+object QueryDetail {
+  def fromQueryExecutionListener(qe: QueryExecution, durationNs: Long): QueryDetail = {
+    val durationMs = TimeUnit.MILLISECONDS.convert(durationNs, TimeUnit.NANOSECONDS)
+    QueryDetail(qe, AtlasUtils.issueExecutionId(), durationMs, Option(SQLQuery.get()))
+  }
+
+  def fromStreamingQueryListener(qe: StreamExecution, event: QueryProgressEvent): QueryDetail = {
+    QueryDetail(qe.lastExecution, AtlasUtils.issueExecutionId(), -1, None,
```

Comment:
Why is it -1 here ? StreamingQueryProgress has a `durationMs` field which gives duration for different phases of the SS query that we could use.

Reply:
That is per micro-batch, which doesn't fit to QueryDetail.

Btw, as we assign `spark_process` entity per application, the attribute actually simply doesn't make sense. It only makes sense when `spark_process` is assigned per query. Before we change the unit of `spark_process`, we might be able to remove the attribute or rename & replace with application start timestamp if possible, but let's file a new issue to do it.

Reply:
The duration updated for SQL query (via fromQueryExecutionListener) is also per query correct ?  So the same thing applies there too (e.g. there could be more than one query in an application) ?

Reply:
Yes maybe we can just drop the attribute until we change the unit of `spark_process` as query based one. Also discussed about this with @ankuriitg regarding this. If you also agree I'll make the change here.